Conflicting declaration of localtime_r between ctime and julia.h on mingw64-x86_64

In file included from `src/codegen.cpp:95:0`:

```
src/julia.h:1502:64: error: conflicting declaration of 'tm* localtime_r(const time_t*, tm*)' with 'C' linkage
 DLLEXPORT struct tm* localtime_r(const time_t *t, struct tm *tm);
                                                                ^
In file included from C:/msys64/mingw64/include/c++/5.2.0/ctime:42:0,
                 from C:/msys64/mingw64/include/llvm/Support/FileSystem.h:36,
                 from C:/msys64/mingw64/include/llvm/Object/Binary.h:19,
                 from C:/msys64/mingw64/include/llvm/Object/SymbolicFile.h:17,
                 from C:/msys64/mingw64/include/llvm/Object/ObjectFile.h:18,
                 from C:/msys64/mingw64/include/llvm/DebugInfo/DIContext.h:20,
                 from C:/msys64/mingw64/include/llvm/ExecutionEngine/RuntimeDyld.h:21,
                 from C:/msys64/mingw64/include/llvm/ExecutionEngine/ExecutionEngine.h:18,
                 from C:/msys64/home/Engineer08/packages/julia/src/julia/src/codegen.cpp:21:
C:/msys64/mingw64/x86_64-w64-mingw32/include/time.h:269:34: note: previous declaration with 'C++' linkage
 __forceinline struct tm *__cdecl localtime_r(const time_t *_Time, struct tm *_Tm) {
                                  ^
```

In `julia.h`,
```
1501:#ifdef _OS_WINDOWS_
1502-DLLEXPORT struct tm* localtime_r(const time_t *t, struct tm *tm);
1503-#endif
```

Since `_OS_WINDOWS_` is 1 on mingw64, this definition collides with the definition from `time.h` which is included in `/mingw64/include/c++/x.x.x/ctime`.

By guarding the definition of `localtime_r`  using `_COMPILER_MINGW_`,
```
#ifdef _OS_WINDOWS_
#if !defined(_COMPILER_MINGW_)
DLLEXPORT struct tm* localtime_r(const time_t *t, struct tm *tm);
#endif
#endif
```

we can avoid the conflict.

--- a/src/julia.h
+++ b/src/julia.h
@@ -1498,10 +1498,6 @@ DLLEXPORT int jl_tcp_bind(uv_tcp_t *handle, uint16_t port, uint32_t host, unsign
 
 DLLEXPORT int jl_sizeof_ios_t(void);
 
-#ifdef _OS_WINDOWS_
-DLLEXPORT struct tm* localtime_r(const time_t *t, struct tm *tm);
-#endif
-
 DLLEXPORT jl_array_t *jl_takebuf_array(ios_t *s);
 DLLEXPORT jl_value_t *jl_takebuf_string(ios_t *s);
 DLLEXPORT void *jl_takebuf_raw(ios_t *s);

